From: Dara Adib <daradib@ocf.berkeley.edu>
Date: Wed, 21 Dec 2016 16:55:01 -0500
Subject: Makefile

Upstream Makefile was configured to build on Mac OS X.

Changes:
 - Build against libraries packaged in Debian.
 - Omit indirect libraries to avoid useless dependencies.
 - Append some CFLAGS instead of defining all CFLAGS
   to allow hardening with Debian defaults.
---
 Makefile | 33 ++++++++++++++-------------------
 1 file changed, 14 insertions(+), 19 deletions(-)

diff --git a/Makefile b/Makefile
index a2ac64d..a9bea23 100644
--- a/Makefile
+++ b/Makefile
@@ -17,32 +17,27 @@ RELEASE_CFLAGS  := -Wall -Wno-unknown-pragmas -Werror -Wno-format -O3 -DNDEBUG
 
 DEBUG_LDFLAGS	:= -g
 
-# Change for DEBUG or RELEASE
-CFLAGS	:= -c $(DEBUG_CFLAGS)
-LDFLAGS	:= $(DEBUG_LDFLAGS)
+CFLAGS += -c -Wall -Wno-unknown-pragmas -Wno-format
 
-OPENZWAVE := ../open-zwave/
-LIBMICROHTTPD := -L/usr/local/lib/ -lmicrohttpd
+OPENZWAVE := /usr/include/openzwave
+LIBMICROHTTPD := -lmicrohttpd
 
-INCLUDES := -I $(OPENZWAVE)/cpp/src -I $(OPENZWAVE)/cpp/src/command_classes/ \
-	-I $(OPENZWAVE)/cpp/src/value_classes/ -I $(OPENZWAVE)/cpp/src/platform/ \
-	-I $(OPENZWAVE)/cpp/src/platform/unix -I $(OPENZWAVE)/cpp/tinyxml/ \
-	-I /usr/local/include/
+INCLUDES := -I $(OPENZWAVE) -I $(OPENZWAVE)/command_classes \
+	-I $(OPENZWAVE)/value_classes -I $(OPENZWAVE)/platform \
+	-I $(OPENZWAVE)/platform/unix -I $(OPENZWAVE)/tinyxml
 
 # Remove comment below for gnutls support
 #GNUTLS := -lgnutls
 
 # for Linux uncomment out next three lines
-LIBZWAVE := $(wildcard $(OPENZWAVE)/*.a)
+LIBZWAVE := -lopenzwave
 #LIBUSB := -ludev
-#LIBS := $(LIBZWAVE) $(GNUTLS) $(LIBMICROHTTPD) -pthread $(LIBUSB) -lresolv
+LIBS := $(LIBZWAVE) $(LIBMICROHTTPD) -pthread
 
 # for Mac OS X comment out above 2 lines and uncomment next 5 lines
 #ARCH := -arch i386 -arch x86_64
 #CFLAGS += $(ARCH)
 #LIBZWAVE := $(wildcard $(OPENZWAVE)/cpp/lib/mac/*.a)
-LIBUSB := -framework IOKit -framework CoreFoundation
-LIBS := $(LIBZWAVE) $(GNUTLS) $(LIBMICROHTTPD) -pthread $(LIBUSB) $(ARCH) -lresolv
 
 %.o : %.cpp
 	$(CXX) $(CFLAGS) $(INCLUDES) -o $@ $<
@@ -59,13 +54,13 @@ ifeq ($(LIBZWAVE),)
 	@exit 1
 endif
 
-ozwcp.o: ozwcp.h webserver.h $(OPENZWAVE)/cpp/src/Options.h $(OPENZWAVE)/cpp/src/Manager.h \
-	$(OPENZWAVE)/cpp/src/Node.h $(OPENZWAVE)/cpp/src/Group.h \
-	$(OPENZWAVE)/cpp/src/Notification.h $(OPENZWAVE)/cpp/src/platform/Log.h
+ozwcp.o: ozwcp.h webserver.h $(OPENZWAVE)/Options.h $(OPENZWAVE)/Manager.h \
+	$(OPENZWAVE)/Node.h $(OPENZWAVE)/Group.h \
+	$(OPENZWAVE)/Notification.h $(OPENZWAVE)/platform/Log.h
 
-webserver.o: webserver.h ozwcp.h $(OPENZWAVE)/cpp/src/Options.h $(OPENZWAVE)/cpp/src/Manager.h \
-	$(OPENZWAVE)/cpp/src/Node.h $(OPENZWAVE)/cpp/src/Group.h \
-	$(OPENZWAVE)/cpp/src/Notification.h $(OPENZWAVE)/cpp/src/platform/Log.h
+webserver.o: webserver.h ozwcp.h $(OPENZWAVE)/Options.h $(OPENZWAVE)/Manager.h \
+	$(OPENZWAVE)/Node.h $(OPENZWAVE)/Group.h \
+	$(OPENZWAVE)/Notification.h $(OPENZWAVE)/platform/Log.h
 
 ozwcp:	ozwcp.o webserver.o zwavelib.o $(LIBZWAVE)
 	$(LD) -o $@ $(LDFLAGS) ozwcp.o webserver.o zwavelib.o $(LIBS)
